src/compiler/transformers/module/module.ts

```diff
@@ -488,12 +488,101 @@ namespace ts {
+        }
+
+        function transformImportCallExpressionUMD(node: ImportCall): Expression {
+            // (function (factory) {
+            //      ... (regular UMD)
+            // }
+            // })(function (require, exports, useSyncRequire) {
+            //      "use strict";
+            //      Object.defineProperty(exports, "__esModule", { value: true });
+            //      require.length === 1 ?
+            //          /*CommonJs Require*/ Promise.resolve().then(() => require('blah'));
+            //          /*Amd Require*/ new Promise(resolve => require(['blah'], resolve));
+            // });
+            const require = createIdentifier("require");
+            return createConditional(
+                /*condition*/ createBinary(createPropertyAccess(require, /*name*/ "length"), /*operator*/ createToken(SyntaxKind.EqualsEqualsEqualsToken), createNumericLiteral("1")),
+                /*whenTrue*/ transformImportCallExpressionCommonJS(node),
+                /*whenFalse*/ transformImportCallExpressionAMD(node)
+            );
+        }
+
+        function transformImportCallExpressionAMD(node: ImportCall): Expression {
+            // improt("./blah")
+            // emit as
+            // define(["require", "exports", "blah"], function (require, exports) {
+            //     ...
+            //     new Promise(resolve => require(['blah'], resolve));
+            // });
+            const resolve = createIdentifier("resolve");
+            return createNew(
+                createIdentifier("Promise"),
+                /*typeArguments*/ undefined,
+                [createArrowFunction(
+                    /*modifiers*/undefined,
+                    /*typeParameters*/ undefined,
+                    [createParameter(/*decorator*/ undefined, /*modifiers*/ undefined, /*dotDotDotToken*/ undefined, /*name*/ resolve)],
+                    /*type*/ undefined,
+                    createToken(SyntaxKind.EqualsGreaterThanToken),
+                    createCall(createIdentifier("require"), /*typeArguments*/ undefined, [createArrayLiteral(node.arguments), resolve]))
+                ]);
+        }
+
+    function transformImportCallExpressionCommonJS(node: ImportCall): Expression {
+            // import("./blah")
+            // emit as
+            // Promise.resolve().then(() => require("./blah"));
+            // We have to wrap require in then callback so that require is done in asynchronously
+            // if we simply do require in resolve callback in Promise constructor. We will execute the loading immediately
+            return createCall(
+                createPropertyAccess(
+                    createCall(
+                        createPropertyAccess(createIdentifier("Promise"), "resolve"),
+                        /*typeArguments*/ undefined,
+                        /*argumentsArray*/[]
+                    ), "then"),
+                /*typeArguments*/ undefined,
+                [createArrowFunction(

```

Comment:
The module transform now happens last (after the ES2015 transformer has run). Arrow functions are not valid ES5/ES3.